system/updated/updated.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+import hashlib
+import os
+import psutil
+import requests
+import shutil
+import time
+
+from pathlib import Path
+
+from openpilot.common.api import API_HOST
+from openpilot.common.basedir import BASEDIR
+from openpilot.common.run import run_cmd
+from openpilot.common.params import Params
+from openpilot.common.realtime import set_core_affinity
+from openpilot.common.swaglog import cloudlog
+from openpilot.system.hardware import HARDWARE
+from openpilot.system.updated.casync import casync
+from openpilot.system.updated.common import get_consistent_flag, set_consistent_flag
+from openpilot.system.version import BuildMetadata, get_build_metadata, build_metadata_from_dict, is_git_repo
+
+
+UPDATE_DELAY = int(os.environ.get("UPDATE_DELAY", 60))
+
+CHANNELS_API_ROOT = "v1/openpilot/channels"
+
+STAGING_ROOT = os.getenv("UPDATER_STAGING_ROOT", "/data/safe_staging")
+
+FINALIZED = Path(STAGING_ROOT) / "finalized"        # where the casync update is pulled
+CASYNC_TMPDIR = Path(STAGING_ROOT) / "casync_tmp"   # working directory for casync temp files
+
+INIT_FILE = Path(BASEDIR) / ".overlay_init"
+
+
+def get_remote_available_channels() -> list | None:
+  try:
+    return list(requests.get(os.path.join(API_HOST, CHANNELS_API_ROOT)).json())
+  except Exception:
+    cloudlog.exception("fetching remote channels")
+    return None
+
+
+def get_remote_channel_data(channel) -> tuple[BuildMetadata | None, dict | None]:
+  try:
+    data = requests.get(os.path.join(API_HOST, CHANNELS_API_ROOT, channel)).json()
+    return build_metadata_from_dict(data["build_metadata"]), data["manifest"]
+  except Exception:
```

Comment:
catch a more specific exception. `Exception` should be caught only in very specific cases